README.md

```diff
@@ -35,7 +35,15 @@ The imported items can then be used elsewhere in your workflow.
 
 ## Inputs
 
-**Device ID:** Generate a device ID with `head -c 16 /dev/urandom | base32 | tr -d = | tr '[:upper:]' '[:lower:]'`. This should be stable across multiple workflow runs. If you're using GitHub-hosted runners you can set this as a secret, and if you're using self-hosted runners you could set it as an env var on the runner.
+**Device ID:** Generate a device ID with `head -c 16 /dev/urandom | base32 | tr -d = | tr '[:upper:]' '[:lower:]'`. This should be stable across multiple workflow runs. If you're using GitHub-hosted runners you can set this as a secret, and if you're using self-hosted runners you could set it as an `OP_DEVICE` environment variable in the runner's shell. In the latter case you'll need to add a step before this action that plucks out the device ID and sets it in the workflow's environment, like:
+
+```yaml
+- name: Get 1Password device ID from macOS runner shell
+  shell: /bin/bash -l {0}
+  run: |
+    echo "OP_DEVICE=$OP_DEVICE" >> $GITHUB_ENV
+    echo "::add-mask::$OP_DEVICE"
```

Comment:
Just so I understand - you are setting the `OP_DEVICE` environment variable here, to override what you are setting in the https://github.com/RobotsAndPencils/1password-action/blob/7361d5ac342bed3a62437441931e009801c86cf2/src/main.ts#L30 

The code that did the fall back is gone, so I'm just making sure that this example code isn't getting overridden by the code above?

Reply:
The short explanation is that this has to happen outside the action itself, which is why there's no change to main.ts anymore, and why there's this new step that might have to be added to workflows beforehand.

Depending on how you set an environment variable on the runner, it might not be automatically available to the workflow. That's the case if you set OP_DEVICE in a file like .bash_profile, .zshenv, etc. So you need to do something to get it from that shell environment on the runner and into the workflow environment. This step does that, by running a Bash login shell, and using the workflow commands to add it to the workflow environment and mask it as a secret.

This is not related to what's happening in main.ts. That code takes the device ID input variable and makes sure its set in the environment when the action invokes the `op` CLI tool, which otherwise doesn't inherit the action or workflow's environment. It's just shuffling the value around to where it needs to be.

This is all specific to self-hosted runners, and I wish it were easier, but this is the best option I've seen so far. It at least makes it possible to uniquely identify self-hosted runners instead of treating all runners, self-hosted or not, as the same machine from `op`'s POV.

Reply:
ah the env is op specific, make sense now.